### PR TITLE
hwChan ordering with unrecorded channels

### LINKED ISSUE #121: Binary file indexing with un-recorded channels.
Migrating from discussion [here](https://github.com/NeuralEnsemble/python-neo/issues/1215#issuecomment-2997606769)

**Issue**
- Currently we index the `hwChan` order assuming all channels from the chip are recorded in the `.rec` file
- Trodes user can choose to not save all channels on the chip to the rec file (e.g.dead contacts)
- This removes the channel from the order of data in the byte stream
- This can shift the index position of a given `hwChan` in the byte stream, leading to errors

**Solution**
When defining the `hwChan` order in the byte stream, intersect the list of channels  inffered wth those included in the rec header

### PR DESCRIPTION
**Pull request by samuelbray32**
Fixes #121 

When making the list of `hwChan` as ordered in the binary stream, restrict to only channels included in the recording.  This ensures proper indexing into the binary data

### CODE REVIEW

File: src/trodes_to_nwb/tests/test_spikegadgets_io.py [hunk 1]
```diff
@@ -402,30 +406,58 @@ def test_produce_ephys_channel_ids():
...
     # --- Edge Cases ---
-    result_5 = SpikeGadgetsRawIO._produce_ephys_channel_ids(0, 32)
+    result_5 = SpikeGadgetsRawIO._produce_ephys_channel_ids(0, 0, 32)
     assert result_5 == []
-    result_6 = SpikeGadgetsRawIO._produce_ephys_channel_ids(128, 0)
+    result_6 = SpikeGadgetsRawIO._produce_ephys_channel_ids(128, 128, 0)
     assert result_6 == []
-    result_7 = SpikeGadgetsRawIO._produce_ephys_channel_ids(0, 0)
+    result_7 = SpikeGadgetsRawIO._produce_ephys_channel_ids(0, 0, 0)
     assert result_7 == []
```
Comment (rly) [inline src/trodes_to_nwb/tests/test_spikegadgets_io.py:455]: Super minor but for consistency, I would update these variable numbers since there is a new `result_5` inserted above.

File: src/trodes_to_nwb/spike_gadgets_raw_io.py [hunk 2]
```diff
@@ -123,7 +132,19 @@ def _produce_ephys_channel_ids(
...
             )
-        return [item for sublist in x for item in sublist]
+
+        channel_names = [item for sublist in x for item in sublist]
+
+        if n_total_channels == n_channels_recorded:
+            # case where all channels are recorded, no censoring required
+            return channel_names
+
+        if hw_channels_recorded is None:
```
Comment (rly) [inline src/trodes_to_nwb/spike_gadgets_raw_io.py:142]: It would be good to add a test for this. I would actually be more specific and use something like
```suggestion
        if not hw_channels_recorded or len(hw_channels_recorded) != n_channels_recorded:
```

**Comment by rly** [comment]
I made some minor suggestions but otherwise looks good to me.

**Review by rly — APPROVED** [review]
